processor/k8sprocessor/kube/client.go

```diff
@@ -146,6 +167,30 @@ func (c *WatchClient) handlePodDelete(obj interface{}) {
 	observability.RecordPodTableSize(int64(podTableSize))
 }
 
+func (c *WatchClient) handleNamespaceAdd(obj interface{}) {
+	if namespace, ok := obj.(*api_v1.Namespace); ok {
+		c.addOrUpdateNamespace(namespace)
+	} else {
+		c.logger.Error("object received was not of type api_v1.Namespace", zap.Any("received", obj))
+	}
+}
+
+func (c *WatchClient) handleNamespaceUpdate(old, new interface{}) {
+	if namespace, ok := new.(*api_v1.Namespace); ok {
+		c.addOrUpdateNamespace(namespace)
+	} else {
+		c.logger.Error("object received was not of type api_v1.Namespace", zap.Any("received", new))
+	}
+}
+
+func (c *WatchClient) handleNamespaceDelete(obj interface{}) {
+	if namespace, ok := obj.(*api_v1.Namespace); ok {
+		c.forgetNamespace(namespace)
```

Comment:
if `forgetNamespace` is being called only from here and just removes a key from a map, probably the code can be inlined here directly.